src/core/ext/filters/client_channel/lb_policy/ring_hash/ring_hash.cc

```diff
@@ -617,41 +652,41 @@ void RingHash::RingHashSubchannelData::UpdateConnectivityStateLocked(
         ConnectivityStateName(last_connectivity_state_),
         ConnectivityStateName(connectivity_state));
   }
-  // Decide what state to report for aggregation purposes.
+  // Decide what state to report for the purposes of aggregation and
+  // picker behavior.
   // If we haven't seen a failure since the last time we were in state
   // READY, then we report the state change as-is.  However, once we do see
   // a failure, we report TRANSIENT_FAILURE and do not report any subsequent
   // state changes until we go back into state READY.
-  if (!seen_failure_since_ready_) {
-    if (connectivity_state == GRPC_CHANNEL_TRANSIENT_FAILURE) {
-      seen_failure_since_ready_ = true;
-    }
-    subchannel_list()->UpdateStateCountersLocked(last_connectivity_state_,
-                                                 connectivity_state);
-  } else {
-    if (connectivity_state == GRPC_CHANNEL_READY) {
-      seen_failure_since_ready_ = false;
-      subchannel_list()->UpdateStateCountersLocked(
-          GRPC_CHANNEL_TRANSIENT_FAILURE, connectivity_state);
+  if (last_connectivity_state_ == GRPC_CHANNEL_TRANSIENT_FAILURE) {
+    // If not transitioning to READY, ignore the update, since we want
+    // to continue to consider ourselves in TRANSIENT_FAILURE.
+    if (connectivity_state != GRPC_CHANNEL_READY) return;
+  } else if (connectivity_state == GRPC_CHANNEL_TRANSIENT_FAILURE) {
+    // If we go from READY to TF, treat it as IDLE.
```

Comment:
nit: as per our offline chat, can we add more to this comment about why we're transitioning to IDLE (e.g. consistency with other langs and better affinity in case the connection just periodically breaks) ?